docs/troubleshooting.rst

```diff
@@ -233,6 +233,30 @@ passed before the ``mocker`` fixture to ensure this:
       # works correctly
       mocker.patch("builtins.open", mocker.mock_open(read_data="content"))
 
+Pathlib.Path objects created outside of tests
+---------------------------------------------
+An pattern which is more often seen with the increased usage of `pathlib` is the
+creation of global `pathlib.Path` objects (instead of string paths) that are used
+in the tests. As these objects are created in the real filesystem, they are not of the same
+type as faked `pathlib.Path` objects, and both will always compare as not equal,
+regardless of the path they point to:
+
+.. code:: python
+
+  import pathlib
+
+  FOLDER = pathlib.Path(__file__).parent
+  FILE = FOLDER / "file.csv"
+
+
+  def test_one(fs):
+      fake_file = pathlib.Path(fs.create_file(FILE).path)
+      assert FILE == fake_file  # fails, compares different objects
+      assert str(FILE) == str(fake_file)  # succeeds, compares the actual paths
```

Comment:
```suggestion
  import pathlib

  # This Path was made in the real filesystem, before the test
  # stands up the fake filesystem
  FILE_PATH = pathlib.Path(__file__).parent / "file.csv"


  def test_path_equality(fs):
      # This Path was made after the fake filesystem is set up,
      # and thus patching within pathlib is in effect
      fake_file_path = pathlib.Path(fs.create_file(FILE_PATH).path)
      assert FILE_PATH == fake_file_path  # fails, compares different objects
      assert str(FILE_PATH) == str(fake_file_path)  # succeeds, compares the actual paths
```

Just cleaning up the snippet I provided:
- No need for `FOLDER`
- Removing room for confusion of `FILE` vague name:
    - `FILE` sounds like a file object when it's a `Path` object
    - Could detail if it's made in real or fake filesystem
- `test_one` name can be more specific